Evaluator spend project (#3893)
langchain-ai/docs: src/langsmith/evaluator-spend.mdx

---
title: Track and limit evaluator spend
sidebarTitle: Evaluator spend
description: Cap weekly LLM spend on evaluators with an organization-wide default or per-attachment limits to keep evaluator costs predictable.
keywords: ["evaluator spend", "spend limit", "evaluator budget", "evaluator cost", "weekly limit", "evaluator monitoring", "stop evaluator", "pause evaluator", "run rule limit"]
---

> Cap weekly LLM spend per evaluator so a runaway evaluator cannot blow your budget. LangSmith tracks evaluator spend in a rolling 7-day window and lets organization admins set a weekly cap on each evaluator's [attachments](/langsmith/evaluation-concepts#attachments), the projects and datasets the evaluator runs on. The cap can be a single organization-wide default or a custom override on a specific attachment.

Reviewer: ```suggestion
Cap weekly LLM spend per evaluator to prevent a single evaluator from exceeding your budget. LangSmith tracks evaluator spend in a rolling 7-day window and lets [organization admins](/langsmith/rbac#organization-admin) set a weekly cap on each evaluator's [attachments](/langsmith/evaluation-concepts#attachments), the projects and datasets the evaluator runs on. The cap can be a single organization-wide default or a custom override on a specific attachment.
```

Author: thanks for catching this AI slop! updated

> LangSmith offers cost controls at three levels: [per-trace and per-model cost tracking](/langsmith/cost-tracking), [tracing usage limits](/langsmith/administration-overview#usage-limits), and weekly evaluator spend caps. This page covers the third.

Reviewer: What about this instead:

```suggestion
This guide shows you how to view and configure weekly evaluator spend caps.

<Tip>
LangSmith also offers [per-trace and per-model cost tracking](/langsmith/cost-tracking) and [tracing usage limits](/langsmith/administration-overview#usage-limits) for cost control.
</Tip>
```

Author: updated

<Warning>

> Spend limits only enforce against runs whose model has [pricing configured](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry) in LangSmith. Verify model pricing before relying on a limit, otherwise a runaway evaluator on an un-priced model will continue without enforcement.

Reviewer: ```suggestion
Spend limits only enforce against runs whose model has [pricing configured](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry) in LangSmith. Verify model pricing before relying on a limit, otherwise a single evaluator on an un-priced model could continue without enforcement.
```
Maybe that should be "will", like would that definitely happen?

Author: I kept "will" because IIUC it "will continue without enforcement".
updated "runaway" to "single"

| [Evaluators table](#evaluators-table) (Spend, Spend Status) | **Evaluators** in the left sidebar | All workspace members |
| [Projects & Datasets tab](#projects--datasets-tab-on-an-evaluator) | Open an evaluator, **Projects & Datasets** | All workspace members |
| [Organization default spend limit](#set-an-organization-default-spend-limit) | Organization **Settings** > **Usage Configuration** | `organization:manage` required to view and edit |
| [Per-attachment override](#override-the-default-for-an-attachment) | Edit evaluator > **Advanced** > **Spend limit** | All members can view, `organization:manage` required to edit |

<CardGroup cols={2}>
  <Card title="Set your first limit" icon="settings" href="#set-an-organization-default-spend-limit">
    Open organization Settings and define a single weekly cap that applies to every attachment across every workspace in the org.
  </Card>
  <Card title="Override for one attachment" icon="edit" href="#override-the-default-for-an-attachment">
    Customize the limit for a specific project or dataset attached to an evaluator.
  </Card>
</CardGroup>

## View evaluator spend

> Spend appears in three places.

Reviewer: ```suggestion
You can find spend in the following UI locations:
```

Author: updated

---
title: Track and limit evaluator spend
sidebarTitle: Evaluator spend
description: Cap weekly LLM spend on evaluators with an organization-wide default or per-attachment limits to keep evaluator costs predictable.
keywords: ["evaluator spend", "spend limit", "evaluator budget", "evaluator cost", "weekly limit", "evaluator monitoring", "stop evaluator", "pause evaluator", "run rule limit"]
---

> Cap weekly LLM spend per evaluator to prevent a single evaluator from exceeding your budget. LangSmith tracks evaluator spend in a rolling 7-day window and lets [organization admins](/langsmith/rbac#organization-admin) set a weekly cap on each evaluator's [attachments](/langsmith/evaluation-concepts#attachments), the projects and datasets the evaluator runs on. The cap can be a single organization-wide default or a custom override on a specific attachment.

Reviewer 2: it will actually be "week to date" window - so things reset on Monday 12AM UTC. 

```suggestion
Cap weekly LLM spend per evaluator to prevent a single evaluator from exceeding your budget. LangSmith tracks weekly evaluator spend from week to date, resetting at Monday 12AM UTC. It lets [organization admins](/langsmith/rbac#organization-admin) set a weekly cap on each evaluator's [attachments](/langsmith/evaluation-concepts#attachments), the projects and datasets the evaluator runs on. The cap can be a single organization-wide default or a custom override on a specific attachment.
```

Author: updated

This guide shows you how to view and configure weekly evaluator spend caps.

<Tip>
LangSmith also offers [per-trace and per-model cost tracking](/langsmith/cost-tracking) and [tracing usage limits](/langsmith/administration-overview#usage-limits) for cost control.
</Tip>

<Warning>

> Spend limits only enforce against runs whose model has [pricing configured](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry) in LangSmith. Verify model pricing before relying on a limit, otherwise a single evaluator on an un-priced model will continue without enforcement.

Reviewer 2: I would add in front of the paragraph 
```suggestion
Setting spend limits is currently available for OpenAI, Anthropic, and Gemini models, with additional model support coming. Spend limits only enforce against runs amongst supported models that have [pricing configured](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry) in LangSmith. Verify model pricing before relying on a limit; unsupported models cannot be created in evaluators once a limit is set.
```

something like the above - it needs to be a supported model (only 3 right now, with more coming) and have pricing set up.

Author: Updated. General docs practice is not to describe current functionality and not promise future functionality. So I did not add "with additional model support coming".

</Warning>

<Note>

> Spend totals are computed over a rolling 7-day window. The UI labels this window as **this week**.

Reviewer 2: Same as above - week to date 
```suggestion
Spend totals are computed based on the spend week to date. The UI labels this window as **this week**.
```

Author: updated

</Note>

## How enforcement works

> LangSmith records spend after each evaluator run completes, then sums spend across the rolling 7-day window. When the total reaches the effective limit, LangSmith pauses the evaluator on that attachment and resumes automatically once spend drops below the limit. In-flight runs may push the total slightly above the cap before they finalize, so an attachment can briefly overshoot by a small amount.

Reviewer 2: same as above - not rolling window

Author: updated

> The agent and the trace are unaffected. Only the evaluator stops producing scores until spend drops below the limit.

Reviewer 2: ```suggestion
The agent and the trace are unaffected. Only the evaluator stops producing scores until the spend limit resets or a custom override is in place.
```

Reviewer 3: "until the spend limit resets or the limit is manually increased"

Author: Updated

## Spend views and controls

| View | Where to find it | Who can see or change it |
|------|------------------|--------------------------|
| [Evaluators page dashboard](#evaluators-page-dashboard) | **Evaluators** in the left sidebar | All workspace members |
| [Evaluators table](#evaluators-table) (Spend, Spend Status) | **Evaluators** in the left sidebar | All workspace members |
| [Projects & Datasets tab](#projects--datasets-tab-on-an-evaluator) | Open an evaluator, **Projects & Datasets** | All workspace members |
| [Organization default spend limit](#set-an-organization-default-spend-limit) | Organization **Settings** > **Usage Configuration** | `organization:manage` required to view and edit |
| [Per-attachment override](#override-the-default-for-an-attachment) | Edit evaluator > **Advanced** > **Spend limit** | All members can view, `organization:manage` required to edit |

<CardGroup cols={2}>
  <Card title="Set your first limit" icon="settings" href="#set-an-organization-default-spend-limit">

>     Open organization Settings and define a single weekly cap that applies to every attachment across every workspace in the org.

Reviewer 2: ```suggestion
    Open organization Settings and define a single weekly cap that applies to all evaluator attachments to every project / dataset across all workspaces in the org.
```

Author: updated

- **Evaluator spend this week**: Total USD spend across all evaluators, with the change versus the previous week.
- **Evaluator runs this week**: Total run count across all evaluators, with the change versus the previous week.
- **Weekly evaluator spend limit monitoring**: Sorted list of top spenders with a per-attachment progress bar against its `$ spent / $ limit`. The header surfaces the count of attachments that have hit their limit (**Limit hit**) or are **on pace to hit limit**.

Use the **Prev week** and **Next week** controls in the page header to move the 7-day window.

The tracing project or dataset view has an **Evaluators** tab that mirrors these widgets scoped to that attachment, for example, **Daily evaluator spend on this tracing project**.

### Evaluators table

The Evaluators table on the same page includes:

- **Spend (this week)**: Total LLM cost for the evaluator across all attachments over the last 7 days. Sortable. Evaluators that do not call an LLM (for example, code evaluators) show no value.
- **Spend Status**: One of the following:
    - **Under limits**: At least one attachment has a limit, and none are at the cap.

>     - **N limit hit**: One or more attachments have reached their limit and are paused. The number reflects how many attachments are paused.

Reviewer 2: ```suggestion
    - **N limit hit**: The evaluator has reached its limit in one or more project or datasets its attached to. The number reflects how many attachments are paused.
```

Overall, I want to avoid using the word "attachment" as it's not reflected in th UI

Author: updated

>     - **Unlimited**: No attachment has a limit set.

Reviewer 2: ```suggestion
    - **Unlimited**: No limits have been set.
```

Author: updated

### Projects & Datasets tab on an evaluator

Open an evaluator and select the **Projects & Datasets** tab to see per-attachment spend and limits:

- **Spend (this week)**: Total LLM cost for the evaluator for that attachment over the last 7 days.

> - **Current spend**: Progress bar showing spend against the limit over the rolling 7-day window.

Reviewer 2: ```suggestion
- **Current spend**: Progress bar showing spend against the limit since Monday 12AM UTC.
```

Author: updated

1. Navigate to **Evaluators** in the left sidebar and open the evaluator.
1. Click the **Edit evaluator** icon at the top right.
1. Under **Source**, select the specific attachment.
1. Scroll past **Filters** and **Sampling Rate**, then expand **Advanced**.
1. In the **Spend limit** field, set a custom USD amount. The unit is `/ week`.
1. **Save** the evaluator configuration.

The hint text below the field shows whether the current value is the organization default or a custom limit. To revert an override back to the organization default, click **Reset to organization default**.

Members without `organization:manage` see the limit but cannot change it. The read-only view shows one of:

- `Unlimited / week (organization default)`
- `$<amount> / week (organization default)`
- `$<amount> / week (custom limit)`

## When a limit is reached

When an attachment's weekly spend reaches its effective limit:

- LangSmith stops running the evaluator on new runs from that attachment.
- The Evaluators table **Spend Status** column shows **N limit hit**, and the Weekly evaluator spend limit monitoring widget surfaces the affected attachment.

> - Skipped runs are not backfilled. Evaluation resumes automatically on new runs once the rolling 7-day spend falls below the limit.

Reviewer 2: ```suggestion
- Skipped runs are not backfilled. Evaluation resumes automatically on new runs once the spend limit resets or a custom override is in place..
```

Author: updated. Instead of "custom override is in place", I used "or the limit is [manually increased](#override-the-default-for-an-attached-project-or-dataset)" based on @madams0013 previous comment.

## Configure model pricing

Spend tracking and limits depend on per-model pricing stored in LangSmith. Confirm that every model used by your evaluators has pricing configured in [Model pricing](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry) and that runs surface cost via [Cost tracking](/langsmith/cost-tracking).

> If a model has no pricing configured, its runs are not counted toward spend, and a spend limit cannot enforce against it.

Reviewer 2: ```suggestion
When a spend limit is set, evaluators can only be run on supported models (OpenAI, Anthropic, Gemini to start, with more coming down the line), and the models need to have pricing configured. Models without pricing configured cannot be used in evaluators.
```

Author: updated

## Troubleshooting

> **Limit set but the evaluator still ran past it**: Check that the model used by the evaluator has a pricing entry in [Model pricing](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry). Spend on un-priced models is not counted, and limits cannot enforce against zero spend.

Reviewer 2: ```suggestion
**Trouble creating an evaluator**: If a limit is set, evaluators can only be run with a supported model (OpenAI, Anthropic, Gemini to start, with more coming down the line) that has a pricing entry in [Model pricing](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry). Spend on un-priced models is not supported once a limit is set.
```

Something like this above - feel free to edit language

Author: updated. I removed "Spend on un-priced models is not supported once a limit is set." as redundant. I did not include "to start, with more coming down the line" due to docs best practice not to promise future features.

**LangSmith spend does not match my LLM provider invoice**: LangSmith computes spend from the per-model rates configured in [Model pricing](/langsmith/cost-tracking#create-a-new-or-modify-an-existing-model-price-entry), not from your provider's billing. Differences are expected if your provider applies discounts, custom contracts, or model variants you have not added to LangSmith.

> **Limit hit earlier than expected**: The window is a rolling 7-day window, not a calendar week. Spend from any run in the past 7 days counts, including spend from earlier in the previous calendar week.

Reviewer 2: Maybe remove given we are doing calendar week

Author: updated
